apps/sms/views/inbox/js/inbox.js

```diff
@@ -398,8 +398,8 @@ var InboxView = {
     selected.forEach((id) => {
       var thread = Threads.get(+id);
 
-      var markable = thread && !thread.isDraft &&
-        (isRead || !thread.getDraft());
+      var markable = thread && !thread.isDraft && 
+              (isRead || !thread.getDraft()) && !(thread.unreadCount ^ isRead));
```

Comment:
Still indentation issue. See previous code[in pink background]. How it was before :)